File: julia/jl-basic/operators.jl
# Operatos

#=
  Task Arithmetic operators
=#
function show_arithmetic_operators()
println("* Aritmentic operators")
# Addition
result = 5 + 3
println(result)  # Output: 8

# Subtraction
result = 10 - 4
println(result)  # Output: 6

# Multiplication
result = 2 * 6
println(result)  # Output: 12

# Division
result = 15 / 4
println(result)  # Output: 3.75

# Modulus (Remainder)
result = 17 % 5
println(result)  # Output: 2

# Increment
num = 5
num += 1
println(num)  # Output: 6

# Decrement
num = 8
num -= 1
println(num)  # Output: 7
    
end

# Task: Assignment operators:
function show_assignmentc_operators()

println("* Assignment operators")
# Assignment operator
x = 10
println(x)  # Output: 10

# Compound assignment operators
y = 5
y += 2
println(y)  # Output: 7

z = 10
z *= 3
println(z)  # Output: 30

w = 15
w /= 2
println(w)  # Output: 7.5

m = 10
m %= 3
println(m)  # Output: 1

end

#Task: Comparison operators
function show_comparison_operator()

println("* Comparison operators")
# Equality
result = (5 == 5)
println(result)  # Output: True

# Inequality
result = (7 != 3)
println(result)  # Output: True

# Greater than
result = (10 > 8)
println(result)  # Output: True

# Less than
result = (6 < 4)
println(result)  # Output: False

# Greater than or equal to
result = (5 >= 5)
println(result)  # Output: True

# Less than or equal to
result = (3 <= 2)
println(result)  # Output: False

end

#Task: Comparison operators
function show_logical_operators()

println("* Logical operators")    
# Logical AND
result = true && false
println(result)  # Output: false

# Logical OR
result = true || false
println(result)  # Output: true

# Logical NOT
result = !true
println(result)  # Output: false

   
end

#Task: Concatenational operator 
function show_concatenation_operator()

println("* Concatenation operator")    
# String concatenation
greeting = "Hello, "
name = "John"
message = greeting * name
println(message)  # Output: Hello, John

end

#Task: Ternary conditional operator
function show_ternary_operator()

    println("* Ternary conditional operator")    
    x = 10
    result = (x % 2 == 0) ? "Even" : "Odd"
    println(result)  # Output: Even
    
 end

show_arithmetic_operators() 
show_assignmentc_operators()
show_comparison_operator()
show_logical_operators()
show_concatenation_operator()
show_ternary_operator()

File: c/src/fundamentals/data-types/operators.c
#include <stdio.h>

int main() {
    // Arithmetic Operators
    int a = 10, b = 3;
    printf("Arithmetic Operators:\n");
    printf("a + b = %d\n", a + b);
    printf("a - b = %d\n", a - b);
    printf("a * b = %d\n", a * b);
    printf("a / b = %d\n", a / b); // Integer division
    printf("a %% b = %d\n", a % b);

    // Comparison Operators
    printf("\nComparison Operators:\n");
    printf("a == b: %d\n", a == b);
    printf("a != b: %d\n", a != b);
    printf("a > b: %d\n", a > b);
    printf("a < b: %d\n", a < b);
    printf("a >= b: %d\n", a >= b);
    printf("a <= b: %d\n", a <= b);

    // Assignment Operators
    printf("\nAssignment Operators:\n");
    a += b; // a = a + b;
    printf("a += b: %d\n", a);
    a -= b; // a = a - b;
    printf("a -= b: %d\n", a);
    // Other assignment operators include *=, /=, %=, <<=, >>=, &=, ^=, |=

    // Logical Operators
    int x = 1, y = 0; // In C, 1 is true and 0 is false
    printf("\nLogical Operators:\n");
    printf("x && y: %d\n", x && y);
    printf("x || y: %d\n", x || y);
    printf("!x: %d\n", !x);

    // Bitwise Operators
    int m = 5, n = 2; // 5 = 0101, 2 = 0010 in binary
    printf("\nBitwise Operators:\n");
    printf("m & n: %d\n", m & n);
    printf("m | n: %d\n", m | n);
    printf("m ^ n: %d\n", m ^ n);
    printf("~m: %d\n", ~m);
    printf("m << 1: %d\n", m << 1);
    printf("m >> 1: %d\n", m >> 1);

    // Increment/Decrement Operators
    int z = 0;
    printf("\nIncrement/Decrement Operators:\n");
    printf("z++: %d\n", z++); // Post-increment
    printf("++z: %d\n", ++z); // Pre-increment
    printf("z--: %d\n", z--); // Post-decrement
    printf("--z: %d\n", --z); // Pre-decrement

    return 0;
}
